### ISSUE TITLE
False-Positive in Lint/DuplicateMethods for dynamic classes

### ISSUE THREAD

**Issue by railsbros-dirk** [bbatsov/rubocop#2726]
I just updated from `0.35.1` to `0.36.0` and received an error for `Lint/DuplicateMethods` in a code similar to this:

```ruby
class Human
  def initialize(attribute = {})
    @attributes = attributes
  end

  def spawn_robot(type)
    type = Class.new do
      def initialize(attributes = {})
        @attributes = attributes
      end
    end
  end
end
```

Running rubocop on this file will give the following error:

```
sandbox/rubocop_test.rb:8:7: W: Lint/DuplicateMethods: Method Human#initialize is defined at both /srv/nfs/fejo-dk/fejo-next/sandbox/rubocop_test.rb:2 and /srv/nfs/fejo-dk/fejo-next/sandbox/rubocop_test.rb:8.
      def initialize(attributes = {})
      ^^^
```

I think this was introduced by the improvements in 1f9c639c574d07b73ceb85db0fc9a76867aaa99e. I'm well aware that we can argue about the code itself but I don't think this should be a `Lint/DuplicateMethods` violation since they are both valid.

I'm not sure how to fix this thus I'm just reporting ;-)

**Comment by bbatsov** [comment]
This was fixed on `master`.

**Comment by bbatsov** [comment]
My bad - I was thinking of a similar problem https://github.com/bbatsov/rubocop/issues/2707

**Comment by alexdowad** [comment]
This is fixed in my open PR.

**Comment by alexdowad** [comment]
This is fixed on `master` now.

**Comment by jonas054** [comment]
Confirmed. The fix was released in 0.37.1.
